### PR TITLE
fix warnings after Elixir version 1.3 and add `mix.lock` and `.gitignore`

### PR DESCRIPTION
**Pull request by melpon**
Elixir 1.4.2 で、現在 issue に上がっている #8 を除いて `mix test` が通ることを確認しています。

- 警告が出ていたので直しました
- `git status` 等を実行した時に `_build/` や `deps/` が変更に出てきて邪魔だったので、`mix new` で自動生成したプロジェクトに入っている `.gitignore` をコピーしておきました。
- `mix.lock` ファイルが無かったので追加しておきました。`mix.lock` ファイルは、依存しているライブラリのバージョンを固定するために必要なので、付けておいた方が良いと思います。
- `test/` の警告は直していません

### CODE REVIEW

**Comment by ma2gedev** [comment]
Hello, @melpon! This is your first Pull Request that will be reviewed by [Ebert](https://ebertapp.io/), an automatic Code Review service. It will leave comments on this diff with potential issues and style violations found in the code as you push new commits. You can also see all the issues found on this Pull Request on its [review page](https://ebertapp.io/github/ma2gedev/power_assert_ex/pulls/10). Please check our [documentation](https://ebertapp.io/docs) for more information.

File: lib/power_assert/ast.ex [hunk 1]
```diff
@@ -12,17 +12,23 @@ defmodule PowerAssert.Ast do
...
+    {form, acc} =
+      unless is_atom(form) do
+        {form, acc} = pre.(form, acc)
+        do_traverse(form, acc, pre, post)
+      else
+        {form, acc}
+      end
+
+    {args, acc} =
+      unless is_atom(args) do
```
Comment (ma2gedev) [inline lib/power_assert/ast.ex]: Unless conditions should not have an `else` block.

File: lib/power_assert/ast.ex [hunk 2]
```diff
@@ -12,17 +12,23 @@ defmodule PowerAssert.Ast do
...
-    end  
-
-    unless is_atom(args) do
-      {args, acc} = Enum.map_reduce(args, acc, fn x, acc ->
-        {x, acc} = pre.(x, acc) 
-        do_traverse(x, acc, pre, post) 
-      end) 
-    end  
+    {form, acc} =
+      unless is_atom(form) do
```
Comment (ma2gedev) [inline lib/power_assert/ast.ex]: Unless conditions should not have an `else` block.

File: lib/power_assert/assertion.ex [hunk 3]
```diff
@@ -494,23 +498,30 @@ defmodule PowerAssert.Assertion do
...
+        str <> "\nonly in rhs: " <> inspect(in_right)
+      else
+        str
+      end
     diff = collect_map_diff(left, right)
-    unless Enum.empty?(diff) do
-      str = str <> "\ndifference:\n" <> Enum.join(diff, "\n")
-    end
+    str =
+      unless Enum.empty?(diff) do
```
Comment (ma2gedev) [inline lib/power_assert/assertion.ex]: Unless conditions should not have an `else` block.

File: lib/power_assert/assertion.ex [hunk 4]
```diff
@@ -494,23 +498,30 @@ defmodule PowerAssert.Assertion do
...
-      str = str <> "\nonly in rhs: " <> inspect(in_right)
-    end
+    str =
+      unless Map.size(in_left) == 0 do
+        str <> "\nonly in lhs: " <> inspect(in_left)
+      else
+        str
+      end
+    str =
+      unless Map.size(in_right) == 0 do
```
Comment (ma2gedev) [inline lib/power_assert/assertion.ex]: Unless conditions should not have an `else` block.

File: lib/power_assert/assertion.ex [hunk 5]
```diff
@@ -494,23 +498,30 @@ defmodule PowerAssert.Assertion do
...
     in_right = Map.split(right, Map.keys(left)) |> elem(1)
     str = "\n"
-    unless Map.size(in_left) == 0 do
-      str = str <> "\nonly in lhs: " <> inspect(in_left)
-    end
-    unless Map.size(in_right) == 0 do
-      str = str <> "\nonly in rhs: " <> inspect(in_right)
-    end
+    str =
+      unless Map.size(in_left) == 0 do
```
Comment (ma2gedev) [inline lib/power_assert/assertion.ex]: Unless conditions should not have an `else` block.

File: lib/power_assert/assertion.ex [hunk 6]
```diff
@@ -494,23 +498,30 @@ defmodule PowerAssert.Assertion do
     |> Enum.join("\n")
   end
   defp extra_information(left, right) when is_map(left) and is_map(right) do
-    if Map.has_key? left, :__struct__ do
-      left = Map.from_struct(left)
-      right = Map.from_struct(right)
-    end
+    left = left |> Map.delete(:__struct__)
+    right = right |> Map.delete(:__struct__)
```
Comment (ma2gedev) [inline lib/power_assert/assertion.ex]: Use a function call when a pipeline is only one function long

File: lib/power_assert/assertion.ex [hunk 7]
```diff
@@ -494,23 +498,30 @@ defmodule PowerAssert.Assertion do
     |> Enum.join("\n")
   end
   defp extra_information(left, right) when is_map(left) and is_map(right) do
-    if Map.has_key? left, :__struct__ do
-      left = Map.from_struct(left)
-      right = Map.from_struct(right)
-    end
+    left = left |> Map.delete(:__struct__)
```
Comment (ma2gedev) [inline lib/power_assert/assertion.ex]: Use a function call when a pipeline is only one function long

File: lib/power_assert/assertion.ex [hunk 8]
```diff
@@ -466,18 +469,19 @@ defmodule PowerAssert.Assertion do
...
-      line = String.replace(line, ~r/\|$/, value)
-      lines = lines ++ [line]
-    end
+    lines =
+      if latest_pos != -1 && latest_pos - (pos + value_len) > 0 do
+        [last_line|tail_lines] = lines |> Enum.reverse
+        {before_str, after_str} = String.split_at(last_line, pos)
+        {_removed_str, after_str} = String.split_at(after_str, value_len)
+        line = before_str <> value <> after_str
+        [line|tail_lines] |> Enum.reverse
```
Comment (ma2gedev) [inline lib/power_assert/assertion.ex]: Use a function call when a pipeline is only one function long

File: lib/power_assert/assertion.ex [hunk 9]
```diff
@@ -466,18 +469,19 @@ defmodule PowerAssert.Assertion do
...
-      lines = [line|tail_lines] |> Enum.reverse
-    else
-      line = String.duplicate(" ", pos + 1)
-      line = replace_with_bar(line, values)
-      line = String.replace(line, ~r/\|$/, value)
-      lines = lines ++ [line]
-    end
+    lines =
+      if latest_pos != -1 && latest_pos - (pos + value_len) > 0 do
+        [last_line|tail_lines] = lines |> Enum.reverse
```
Comment (ma2gedev) [inline lib/power_assert/assertion.ex]: Use a function call when a pipeline is only one function long

**Comment by ma2gedev** [comment]
Ebert has finished reviewing this Pull Request and has found:

* 8 possible new issues (including those that may have been commented here).
* 1 fixed issue! 🎉

You can see more details about this review at https://ebertapp.io/github/ma2gedev/power_assert_ex/pulls/10.

**Comment by ma2gedev** [comment]
Ebert has finished reviewing this Pull Request and has found:

* 3 possible new issues (including those that may have been commented here).
* 2 fixed issues! 🎉

You can see more details about this review at https://ebertapp.io/github/ma2gedev/power_assert_ex/pulls/10.

**Review by ma2gedev — APPROVED** [review]

**Comment by ma2gedev** [comment]
Ebert has finished reviewing this Pull Request and has found:

* 3 fixed issues! 🎉

You can see more details about this review at https://ebertapp.io/github/ma2gedev/power_assert_ex/pulls/10.

**Comment by melpon** [comment]
あ、既に https://github.com/ma2gedev/power_assert_ex/tree/fix-warnings-after-1.3 で直していたのに気が付きませんでした。
`.gitignore` と `mix.lock` 追加するのだけ出し直します！

**Comment by ma2gedev** [comment]
@melpon Thank you for your pull request 😄 

Thanks for fixing warnings, adding `.gitignore` and `mix.lock` that are helpful.
So I will re-open this pull request and merge. 

Please do not mind my branch fix-warnings-after-1.3, I am worried because I did it while I was sleepy. 😱

**Review by ma2gedev — APPROVED** [review]

**Comment by melpon** [comment]
Thank you for merging my pull request!

**Comment by ma2gedev** [comment]
Just published v0.1.0 to hex! https://hex.pm/packages/power_assert

**Pull request merged**